packages/modules/providers/file-s3/src/services/s3-file.ts

```diff
@@ -175,4 +178,49 @@ export class S3FileService extends AbstractFileProviderService {
       expiresIn: this.config_.downloadFileDuration,
     })
   }
+
+  // Note: Some providers (eg. AWS S3) allows IAM policies to further restrict what can be uploaded.
+  async getPresignedUploadUrl(
+    fileData: FileTypes.ProviderGetPresignedUploadUrlDTO
+  ): Promise<FileTypes.ProviderFileResultDTO> {
+    if (!fileData) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `No file data provided`
+      )
+    }
+
+    if (!fileData.filename) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `No filename provided`
+      )
+    }
+
+    const parsedFilename = path.parse(fileData.filename)
+    const fileKey = `${this.config_.prefix}${parsedFilename.name}${parsedFilename.ext}`
```

Comment:
```suggestion
    const fileKey = `${this.config_.prefix}${parsedFilename.base}`
```

Reply:
👍 Initially I thought it's better to separate them in case we want to apply some suffix to the filename, but we can do it later on if necessary